config/routes.rb

```diff
@@ -5,6 +5,7 @@
   resources :merchants, only: [:show] do
     resources :items, only: [:index]
     resources :invoices, only:[:index]
+    get '/invoices/:id', to: 'merchants/invoices#show'
```

Comment:
We could also try utilizing resources here, with the controller: "/merchants/invoices" syntax